sound/soc/msm/msm8x16.c

```diff
@@ -542,16 +540,18 @@ static int lineout_status_put(struct snd_kcontrol *kcontrol,
 	pr_debug("%s: external speaker PA mode:%d\n", __func__, state);
 
 	switch (state) {
+	case 0:
+		schedule_delayed_work(&lineout_amp_disable, msecs_to_jiffies(50));
+		break;
 	case 1:
 		schedule_delayed_work(&lineout_amp_enable, msecs_to_jiffies(50));
 		break;
-	case 0:
-		//schedule_delayed_work(&lineout_amp_disable, msecs_to_jiffies(5));
-		msm8x16_ext_spk_control(0);
-		break;
 	case 2:
 		schedule_delayed_work(&lineout_amp_dualmode, msecs_to_jiffies(50));
 		break;
+	case 3:
```

Comment:
Can you just try without the case 3 I'm pretty sure it doesn't do anything.

SOC_ENUM_EXT("Lineout_1 amp", msm_snd_enum[3],
			lineout_status_get, lineout_status_put),
This defines the actions to do when Lineout_1 amp is set by the audio. It calls lineout_status_get or lineout_status_put and possible values are defined in msm_snd_enum[3]. which is 
SOC_ENUM_SINGLE_EXT(3, lineout_text) =>  char *const lineout_text[] = {"DISABLE", "ENABLE", "DUALMODE"};

I was debugging the issue from the kernel side + mixer_path file, and I managed to fix it playing with the Lineout_amp 1 but fix was mush more intrusive. I actually lose all this debugging by a stupid repo sync --force

But the thing is I'm pretty sure that the value can never be 3.